### PR TITLE
Use a private class + static instance for proxy equality comparers

### PR DESCRIPTION
**Pull request by pepone**
This small PR replaces the public structs that implement proxy IEqualityComparer by private classes + public static instances in Proxy class

### CODE REVIEW

**Comment by pepone** [comment]
> Looks good.
> 
> I am wondering if it would be even better to create a separate `ProxyComparer` class modeled after `StringComparer`, i.e.:

Reworked this to model it after StringComparer

File: csharp/src/Ice/ProxyComparer.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+using System.Collections.Generic;
+
+namespace ZeroC.Ice
+{
+    /// <summary>Represents a <see cref="IObjectPrx">object proxy</see>comparison operation that uses specific rules.
```
Comment (bernardnormier) [inline csharp/src/Ice/ProxyComparer.cs:9]: What about:
Represents an object proxy comparison operation based on all or only some of the proxy properties. The Default property delegates to the implementation of IEquatable<T> provided by IObjectPrx.

(and no need to describe the other fields in this summary).

File: csharp/src/Ice/ProxyComparer.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                {
+                    return false;
+                }
+
+                return lhs.Identity.Equals(rhs.Identity) && lhs.Facet.Equals(rhs.Facet);
+            }
+        }
+
+    }
+}
```
[the remark this answers is not in the document]
Comment (pepone) [reply csharp/src/Ice/ProxyComparer.cs]: fixed

File: csharp/src/Ice/ProxyComparer.cs (hunk 1 above)
Comment (pepone) [reply csharp/src/Ice/ProxyComparer.cs]: fixed

**Pull request merged**